### ISSUE TITLE
Have `Home`/`End` keys change cursor position, not selected item

### ISSUE THREAD

**Issue by bhollis** [downshift-js/downshift#1162]
Downshift (`useComboBox`) is set up for `Home` and `End` keys to select the first or last option in the dropdown. However, the [WAI-ARIA ListBox spec](https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html) suggests that those keys should have their normal behavior of moving the cursor to the beginning/end of the text input.

**Comment by bhollis** [comment]
<img width="768" alt="Screen Shot 2020-08-25 at 2 09 29 PM" src="https://user-images.githubusercontent.com/313208/91228398-05577500-e6dd-11ea-8827-3b85e378fedd.png">

**Comment by bhollis** [comment]
This has been a common complaint from our users as we introduce Downshift-based combobox UIs - that they would prefer to continue using Home/End as they are used to.

**Comment by silviuaavram** [comment]
```
Home (Optional): Either moves focus to and selects the first option or returns focus to the textbox and places the cursor on the first character.
End (Optional): Either moves focus to the last option or returns focus to the textbox and places the cursor after the last character.
```

This is from https://www.w3.org/TR/wai-aria-practices/#listbox-popup-keyboard-interaction. Yours is from an implementation that chose the second approach. So neither of us is wrong.

I would prefer to leave it like it is, and you can use `preventDownshiftDefault` on Home/End (https://github.com/downshift-js/downshift#customizing-handlers) and have your scenario fixed.

If I make it as standard then there will be no way to highlight first/last element for the ones that chose this approach, unless they implement it themselves.

Let me know if you have further questions.

**Comment by bhollis** [comment]
Makes sense - I've already applied that fix. It may be a difference between combobox and select mode - I would expect all the normal text navigation commands to continue to work as long as I have an input field I'm typing into.

**Comment by silviuaavram** [comment]
Will close this, since we already have a workaround and details posted.

**Comment by joaopaulobdac** [comment]
@silviuaavram Wouldn't it be less annoying if the home/end keys are only hijacked when the box is open and interactive? Currently, even if you have a selected item in the input you still can't move the cursor using home/end. Also relevant for #1097

**Comment by silviuaavram** [comment]
@joaopaulobdac re-opened that the ticked and posted my opinion on it. Can you take a look and maybe create a PR to fix it? Thanks!

**Comment by okonet** [comment]
>  It may be a difference between combobox and select mode - I would expect all the normal text navigation commands to continue to work as long as I have an input field I'm typing into.

I agree with this. And the recommendation says the same:

> Either moves focus to and selects the first option or, if the combobox is editable, returns focus to the combobox and places the cursor on the first character.

So, its either selection _or its editable_ moves the cursor. At least that's how I read it. So it's not a simple _or_ but rather an _if_. WDYT?

**Comment by silviuaavram** [comment]
I will reopen this ticket. The proposal is to change the behaviour only for useCombobox. useSelect works as expected, and Downshift has been like this since forever, so we won't change that moving forward.

Thanks everyone for your input!
